minimint/src/consensus/mod.rs

```diff
@@ -342,6 +402,13 @@ where
             )
             .collect();
 
+        if let Some(epoch) = self.db.get_value(&LastEpochKey).unwrap() {
+            let last_epoch = self.db.get_value(&epoch).unwrap().unwrap();
+            let sig = self.cfg.hbbft_sks.0.sign(last_epoch.hash);
+            let item = ConsensusItem::EpochInfo(EpochSignatureShare(sig));
+            items.push(item);
+        };
```

Comment:
This means each epoch will only ever be signed by `t` instead of `n` guardians because if our signature isn't part of the consensus we won't try to resubmit. This can afaik allow an adversary to produce an epoch that isn't sufficiently signed.

If the adversary is part of the consensus outcome but didn't contribute they get banned, but the honest guardian not part of the consensus outcome last round does not contribute the missing signature retroactively.